### PR TITLE
Idle and Attack Animations

### PR DESCRIPTION
**Pull request by darylnak**

### CODE REVIEW

File: client/Window.cpp [hunk 1]
```diff
@@ -179,27 +179,41 @@ bool Window::initializeObjects()
...
-	#ifdef SERVER_ENABLED
-	clientChar = chars[client->getId()];
-
-	#else 
 	clientChar = chars[0];
-	gameStarted = true;
+	Window::gameStarted = true;
 	#endif
 
+	chars[3] = (new Character("shaders/character/billboard.obj", &projection, &view, &eyePos, texShader,
```
Comment (siueric0010) [inline client/Window.cpp:211]: Monster (or chars[3]) should have SPAWN_POSITIONS[3][0] for x and SPAWN_POSITIONS[3][1] for z.

In other words, replace 115.f with SPAWN_POSITIONS[3][0] and 5.f with SPAWN_POSITIONS[3][1] in the translation vector (first vec3 of the constructor).

File: client/Window.cpp [hunk 2]
```diff
@@ -431,15 +445,39 @@ void Window::handleRoleClaim(GameUpdate update) {
...
+				"shaders/character/CLERIC"));	
+			chars[update.id]->loadAnimationAssets("shaders/character/CLERIC");
 			break;
 		case ROGUE:
+			guiManager->selectScreen->selectScreenElements[4]->loadTexture("shaders/select_screen/rogue_selected.png");
+			chars[update.id] = (new Character("shaders/character/billboard.obj", &projection, &view, &eyePos, texShader,
+				glm::vec3(70.f, 1.f, 70.f), glm::vec3(0.f, 1.f, 0.f), glm::radians(0.f), 5.f, glm::vec3(1.f, .5f, .5f),
+				"shaders/character/ROGUE"));
+			chars[update.id]->loadAnimationAssets("shaders/character/ROGUE");
 			break;
```
Comment (siueric0010) [inline client/Window.cpp]: A couple things to change here:
1. delete guiManager->selectScreen->selectScreenElements[...]->loadTexture that are inside the switch statement, that's old code
2. the first vec3 (i.e. the translation vector) in the new Character() constructions are dependent on the player's id number. (look at GameConstants.h:115 and Game.cpp's initPlayers(). So you may want to do glm::vec3(SPAWN_POSITIONS[update.id][0], 1.f, SPAWN_POSITIONS[update.id][1]) for each new Character()

File: CppProperties.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (siueric0010) [inline CppProperties.json:1]: Also, idk if this file is required for your code, but maybe gitignore it?

File: CppProperties.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      "defines": [
+        "WIN32",
+        "NDEBUG",
+        "UNICODE",
+        "_UNICODE"
+      ],
+      "intelliSenseMode": "windows-msvc-x64"
+    }
+  ]
+}
```
Comment (siueric0010) [inline CppProperties.json]: Also, idk if this file is required for your code, but maybe gitignore it if we don't need it?

**Review by duouyang — COMMENTED** [review]
common/game code looks good to me!

**Review by siueric0010 — APPROVED** [review]
Tested on my machine, looks good to me. eventually we will get more animations on the drive and then we can add more.

**Pull request merged**